src/macros.jl

```diff
@@ -474,6 +474,245 @@ function parse_constraint_head(
+x
+```
+"""
+op_ifelse(a, x, y) = ifelse(a, x, y)
+
+# We can't make this a generic `NonlinearOperator` because we only want to
+# intercept `ifelse` if the first argument is an `AbstractJuMPScalar` (if it's a
+# `Bool`, we want to return the correct branch).
+op_ifelse(a::AbstractJuMPScalar, x, y) = NonlinearExpr(:ifelse, Any[a, x, y])
+
+"""
+    op_and(x, y)
+
+A function that falls back to `x & y`, but when called with JuMP variables or
+expressions, returns a [`GenericNonlinearExpr`](@ref).
+
+## Example
+
+```jldoctest
+julia> model = Model();
+
+julia> @variable(model, x);
+
+julia> op_and(true, false)
+false
+
+julia> op_and(true, x)
+true && x
+```
+"""
+const op_and = NonlinearOperator(:&&, &)
+# Note that the function is `&` instead of `&&` because `&&` is special lowering
+# syntax and is not a regular Julia function, but the MOI operator is `:&&`.
+
+"""
+    op_or(x, y)
+
+A function that falls back to `x | y`, but when called with JuMP variables or
+expressions, returns a [`GenericNonlinearExpr`](@ref).
+
+## Example
+
+```jldoctest
+julia> model = Model();
+
+julia> @variable(model, x);
+
+julia> op_or(true, false)
+true
+
+julia> op_or(true, x)
+true || x
+```
+"""
+const op_or = NonlinearOperator(:||, |)
+# Note that the function is `|` instead of `||` because `||` is special lowering
+# syntax and is not a regular Julia function, but the MOI operator is `:||`.
+
+"""
+    op_less_than(x, y)
+
+A function that falls back to `x < y`, but when called with JuMP variables or
+expressions, returns a [`GenericNonlinearExpr`](@ref).
+
+## Example
+
+```jldoctest
+julia> model = Model();
+
+julia> @variable(model, x);
+
+julia> op_less_than(1, 2)
+true
+
+julia> op_less_than(x, 2)
+x < 2
+```
+"""
+const op_less_than = NonlinearOperator(:<, <)
+
+"""
+    op_greater_than(x, y)
+
+A function that falls back to `x > y`, but when called with JuMP variables or
+expressions, returns a [`GenericNonlinearExpr`](@ref).
+
+## Example
+
+```jldoctest
+julia> model = Model();
+
+julia> @variable(model, x);
+
+julia> op_greater_than(1, 2)
+false
+
+julia> op_greater_than(x, 2)
+x > 2
+```
+"""
+const op_greater_than = NonlinearOperator(:>, >)
+
+"""
+    op_less_equal(x, y)
+
+A function that falls back to `x <= y`, but when called with JuMP variables or
+expressions, returns a [`GenericNonlinearExpr`](@ref).
+
+## Example
+
+```jldoctest
+julia> model = Model();
+
+julia> @variable(model, x);
+
+julia> op_less_equal(2, 2)
+true
+
+julia> op_less_equal(x, 2)
+x <= 2
+```
+"""
+const op_less_equal = NonlinearOperator(:<=, <=)
+
+"""
+    op_greater_equal(x, y)
+
+A function that falls back to `x >= y`, but when called with JuMP variables or
+expressions, returns a [`GenericNonlinearExpr`](@ref).
+
+## Example
+
+```jldoctest
+julia> model = Model();
+
+julia> @variable(model, x);
+
+julia> op_greater_equal(2, 2)
+true
+
+julia> op_greater_equal(x, 2)
+x >= 2
+```
+"""
+const op_greater_equal = NonlinearOperator(:>=, >=)
```

Comment:
I would have expected `op_greater_or_equal` here

Reply:
Changed to `op_greater_than_or_equal_to`